### ISSUE TITLE
[Q] Streams working on 4G but not behind some firewalls

### ISSUE THREAD

**Issue by sawariz0r** [kevinshen56714/SkyOffice#24]
Hey, thanks for making this. 
It has made our virtual office environment much, much easier to work in.

However - I've had a few problems I haven't had time to debug.
One of them is that some of my employees can't see/stream their video or sound on their home wifi's. 
But on 4G or at the office (well, local network) it works perfectly.

Have you encountered anything like this and if yes, do you have a solution for it?

**Comment by kevinshen56714** [comment]
@sawariz0r thanks for trying out SkyOffice. Currently, we are using the free public PeerServer, which seems to work fine for most users yet has its limitations. We've noticed [an issue similar to yours from other PeerJS users](https://github.com/peers/peerjs/issues/284?fbclid=IwAR2IkQegMd4kyxQcv7vEPIFrq8gHGhO1NpR5nBRo_lJ6hUBsLcNLLeJJuxI). Hosting your own TURN server may be the best solution.

We are currently working on adding some more features to SkyOffice (chat, private rooms, etc.) But in the long run, we plan to give the project a revamp and switch to [Daily.co](https://github.com/daily-co/daily-js). And hopefully, the video connection will be more stable.

**Comment by sawariz0r** [comment]
Yeah, I ended up hosting a TURN server and that resolved the issues. That sounds like a more solid plan in the long run, since it looks like the Daily.co would be quite helpful. Let me know if you need an extra hand, I?m happy to contribute as long as this stays open source. Since it?s been really helpful for both me and my colleagues :)s?n 28 nov. 2021 kl. 04:33 skrev Kuan-Hsuan (Kevin) Shen ***@***.***>:
@sawariz0r thanks for trying out SkyOffice. Currently, we are using the free public PeerServer, which seems to work fine for most users yet has its limitations. We've noticed an issue similar to yours from other PeerJS users. Hosting your own TURN server may be the best solution.
We are currently working on adding some more features to SkyOffice (chat, private rooms, etc.) But in the long run, we plan to give the project a revamp and switch to Daily.co. And hopefully, the video connection will be more stable.

?You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.
